res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,869 @@
+        engine.setValue(group, "reloop_exit", true);
+    }
+};
+
+DenonHC1000S.beatLoopRoll = function(midino, control, value, status, group) {
+    DenonHC1000S.beatroll = !DenonHC1000S.beatroll; // '!' inverts a boolean (true/false) value
+    print("beatroll is pressed " + DenonHC1000S.beatroll);
+    engine.setValue(group, "beatlooproll_activate", DenonHC1000S.beatroll);
+};
+
+DenonHC1000S.pitchBendDown = function(midino, control, value, status, group) {
+    if (DenonHC1000S.shiftSampleLeft || DenonHC1000S.shiftSampleRight) { //with Sample-shift use this button for DVS Ctrl
+        var vinylcontrolStatus = engine.getValue(group, "vinylcontrol_enabled");
+        if ((status & 0xF0) === 0x90) {
+            engine.setValue(group, "vinylcontrol_enabled", (!vinylcontrolStatus)); //DVS VinylControl Toggle (software)
+        }
+    } else {
+        if (!DenonHC1000S.shift) {
+            if ((status & 0xF0) === 0x90) {
+                engine.setValue(group, "rate_temp_down", 1);
+            } else {
+                engine.setValue(group, "rate_temp_down", 0);
+            }
+        } else {
+            if ((status & 0xF0) !== 0x90) {
+                engine.setValue(group, "back", false);
+            } else {
+                engine.setValue(group, "back", true);
+            }
+        }
+    }
+};
+
+DenonHC1000S.pitchBendUp = function(midino, control, value, status, group) {
+    if (DenonHC1000S.shiftSampleLeft || DenonHC1000S.shiftSampleRight) { //with Sample-shift use this button for DVS Ctrl
+        var passthroughStatus = engine.getValue(group, "passthrough");
+        if ((status & 0xF0) === 0x90) {
+            engine.setValue(group, "passthrough", (!passthroughStatus)); //DVS Passthrough (software)
+        }
+    } else {
+        if (!DenonHC1000S.shift) {
+            if ((status & 0xF0) === 0x90) {
+                engine.setValue(group, "rate_temp_up", 1);
+            } else {
+                engine.setValue(group, "rate_temp_up", 0);
+            }
+        } else {
+            if ((status & 0xF0) !== 0x90) {
+                engine.setValue(group, "fwd", false);
+            } else {
+                engine.setValue(group, "fwd", true);
+            }
+        }
+    }
+};
+
+DenonHC1000S.jumpBack = function(midino, control, value, status, group) {
+    if (DenonHC1000S.shiftSampleLeft || DenonHC1000S.shiftSampleRight) { //with Sample-shift use this button for DVS Ctrl
+        var vinylcontrolModeBefore = engine.getValue(group, "vinylcontrol_mode");
+        if (vinylcontrolModeBefore < 2) {
+            var vinylcontrolModeNext = (vinylcontrolModeBefore + 1);
+        } else {
+            vinylcontrolModeNext = 0;
+        }
+        if ((status & 0xF0) === 0x90) {
+            print("DVS changing vinyl control mode to " + vinylcontrolModeNext);
+            engine.setValue(group, "vinylcontrol_mode", vinylcontrolModeNext); //DVS VinylControl Mode
+        }
```

Comment:
this might introduce a bug because depending on how you mapped the function in the XML. The mode might be advanced twice (when the function is also called on button release).